File: 백준/Bronze/13300. 방 배정/방 배정.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int N = Integer.parseInt(st.nextToken());
        int K = Integer.parseInt(st.nextToken());

        int[][] rooms = new int[2][7];

        for (int i = 0; i < N; i++) {
            st = new StringTokenizer(br.readLine());
            int gender = Integer.parseInt(st.nextToken());
            int grade = Integer.parseInt(st.nextToken());
            rooms[gender][grade]++;
        }

        int roomCnt = 0;
        for (int g = 0; g <= 1; g++) {
            for (int y = 1; y <= 6; y++) {
                if (rooms[g][y] > 0) {
                    roomCnt += (rooms[g][y] + K - 1) / K;
                }
            }
        }
        System.out.print(roomCnt);
    }
}

File: 백준/Bronze/13300. 방 배정/방 배정.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val st = StringTokenizer(br.readLine())
    val n = st.nextToken().toInt()
    val k = st.nextToken().toInt()
    val arr = Array(2) { IntArray(7) }
    for (i in 0 until n) {
        val st1 = StringTokenizer(br.readLine())
        val s = st1.nextToken().toInt()
        val y = st1.nextToken().toInt()
        arr[s][y]++
    }
    var cnt = 0
    for (i in 0..6) {
        if (arr[0][i] % k == 0) {
            cnt += arr[0][i] / k
        } else {
            cnt += arr[0][i] / k + 1
        }
        if (arr[1][i] % k == 0) {
            cnt += arr[1][i] / k
        } else {
            cnt += arr[1][i] / k + 1
        }
    }
    println(cnt)
}